### PR TITLE
repair multipoolAutoswap swapOut secondary to secondary

### LINKED ISSUE #3071: Multipool Autoswap - SwapOut - Secondary to Secondary case violates rights conservation
## Describe the bug

When analyzing multipoolAutoswap to ensure that rights were being conserved in all the reallocations, I found that the code in swap.js starting at line 216 was not right. Here's the stagings for the reallocation. The users' seat:

https://github.com/Agoric/agoric-sdk/blob/99a34c6cbd49b076a6965bcda318ae7647a4b106/packages/zoe/src/contracts/multipoolAutoswap/swap.js#L227-L232

The pool for the brandIn:

https://github.com/Agoric/agoric-sdk/blob/99a34c6cbd49b076a6965bcda318ae7647a4b106/packages/zoe/src/contracts/multipoolAutoswap/swap.js#L234-L240

The pool for the brandOut:

https://github.com/Agoric/agoric-sdk/blob/99a34c6cbd49b076a6965bcda318ae7647a4b106/packages/zoe/src/contracts/multipoolAutoswap/swap.js#L242-L251

The amounts in the amountIn brand should be conserved (if we add it to one seat, we must subtract it from another, etc). However, we are adding it to the pool *and* giving it back to the user. That can't be right.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
fixes: #3071

### CODE REVIEW

**Review by katelynsills — COMMENTED** [review]
Had a question about calculations, but otherwise looks good!

File: packages/zoe/test/unitTests/contracts/test-multipoolAutoswap.js [hunk 1]
```diff
@@ -1872,3 +1872,234 @@ test('multipoolAutoSwap collect empty fees', async t => {
...
+  );
+  t.truthy(
+    AmountMath.isEqual(
+      getAmountOut(quotePostTrade),
+      AmountMath.make(25, centralR.brand),
+    ),
+  );
+
+  const moolaPayoutAmount = await moolaR.issuer.getAmountOf(bobMoolaPayout);
+  t.deepEqual(moolaPayoutAmount, moola(201), `bob gets 201 moola`);
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-multipoolAutoswap.js]: Yes.  I changed the request for a price to use `getPriceGivenRequiredOutput` so we can see that the 201 is as expected.

[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/zoe/test/unitTests/contracts/test-multipoolAutoswap.js]: sounds good!

**Review by katelynsills — APPROVED** [review]
LGTM!

**Pull request merged**